### DISCUSSION TITLE
Support to mark and extension as "FREE TRIAL"

### DISCUSSION THREAD

**Discussion by chrisdias** [microsoft/vsmarketplace#145] [Ideas]
the vs code extension marketplace only supports extensions being marked as `FREE`:

![free](https://user-images.githubusercontent.com/1487073/55763843-a593f780-5a1d-11e9-8151-b6b5b2e19bde.png)

Some extensions however are actually `FREE TRIAL`, so it would be good to let an extension declare this so people get a clearer understanding of what they are installing.

The default should always be `FREE`, so this option (I assume in `package.json`) would be to let the publisher override this.

**Comment by chrisdias** [comment]
We don't have any plans on having "for pay" extensions for VS Code, so here are two options for adding this to `package.json`:

* The minimal thing we could do is have a new `boolean` field in the manifest called `trial`.  The default is `false` but if set to `true` then we tell the marketplace its a "Free Trial".  This follows the `preview` model which is `true | false`.

* To be more inline with the rest of the Marketplace we could have something like a `pricingTier` field that is either `free` or `trial` with `free` being the default.  We just don't have a `paid` option.

Thoughts?

**Comment by sandy081** [comment]
@chrisdias How does it useful supporting a trial model without paid model? What are the usecases?

**Reply by ssbarnea** [reply]
By now, you should know that the cloud, aka hostes service, is the future use-case for almost any software. Look at first recent example, Github Copilot.

**Comment by chrisdias** [comment]
The [Devsense PHP extension](https://marketplace.visualstudio.com/items?itemName=DEVSENSE.phptools-vscode) has a "free trial" for their extension. It is free to use the extension for 15 days and then you have to then buy a license through their website.  Once you buy the license, you provide an access key to the extension to keep using it. 

This is the only extension I know of that does this, and it is creating some confusion (see [here](https://marketplace.visualstudio.com/items?itemName=DEVSENSE.phptools-vscode&ssr=false#review-details)).

However, your question raised a good point. The free trial and for pay models in the other parts of the Marketplace are for purchasing the license _through the Marketplace_. In this case, you purchase the extension through a third party market. 

So, if we add this, then in one part of the Marketplace the notion of "FREE TRIAL" means something very different than in another part of the Marketplace. As a result, I don't think we should mix the two models.

**Comment by chrisdias** [comment]
Hmmm, i quickly stand corrected.  In the Visual Studio section of the Marketplace, there are "FREE TRIAL" extensions (e.g. [Arduino](https://marketplace.visualstudio.com/items?itemName=VisualMicro.ArduinoIDEforVisualStudio)) as well that are not purchased through the Marketplace. So, maybe it is OK to support the same thing for VS Code.

So, the Devsense extension is the use case of a "FREE TRIAL" extension.

**Comment by WiliTest** [comment]
I wasted my time with an extension that was presented as "license" (which could mean anything), which was actually free for a few days, then most of its features weren't free anymore. 

In the presentation of this extension, the developer doesn't mention that most features it offers are only available for a trial of a few days. 

My feature request:
Free extensions are mentioned as "free", so paid extension should be mentioned as "paid", and/or "free trial" and/or "limited free version". And, in the VScode extension presentation,  the developers should follow a standard presentation showing which feature are free and which are trial limited. 

And another problem: the extension I'm talking about is mentioned as "free" in the marketplace.visualstudio.com but as "license" in the presentation of the extension in VScode, which is even more confusing.

**Comment by jakubmisek** [comment]
currently we are providing paid extension with trial - we wouldn't mind marking it "free trial" or "trial" or "paid" or whatever - we don't want to confuse our users and we don't want to mystify them :)

https://marketplace.visualstudio.com/items?itemName=DEVSENSE.phptools-vscode

for now, we have marked it `preview` and we added badges all over the readme

![image](https://user-images.githubusercontent.com/842150/64343008-fff4ac80-cfeb-11e9-93cb-6d787c0714ad.png)
![image](https://user-images.githubusercontent.com/842150/64343021-0a16ab00-cfec-11e9-8f8d-b919c451c1fd.png)


If there would be anything else we can do, please advice. Since we're already in contact with Microsoft devs and we know nothing will happen in the near future, we'd be happy for any feedback on what we should acknowledge and where to let people know, it is a paid commercial software.

**Comment by gjsjohnmurray** [comment]
How might this be progressed in 2020?

Folk interested in this might also want to upvote microsoft/vscode#90648 (support for premium extensions)

**Comment by RandomFractals** [comment]
wow! that's all we got on paid extensions? I am sure vsode dev team gets paid for their work.

I'd like to get a coffee once in a while for my bits too ... as I am sure other ext. devs since I still see vscode mostly as a really good IDE shell with OSS APIs & extensions mainly making it a great IDE & dev/ops tool ...

**Comment by RandomFractals** [comment]
not sure why this was moved to a discussion.

@chrisdias any updates on this?

**Comment by DataTraveler1** [comment]
Hi @chrisdias, I was wondering if there might be any updates on this. There is still a demand for a "trial" or "free trial" property for the marketplace.

Can anyone suggest a way to move this topic forward?
